src/Context/PeopleContext.tsx

```
@@ -0,0 +1,53 @@
+import React, { createContext, useContext, useEffect, useState } from 'react';
+import { Person } from '../types/Person';
+import { getPeople } from '../utils/fetchClient';
+import { ErrorMessage } from '../types/ErrorMessage';
+
+type UsersContextType = {
+  people: Person[];
+  setPeople: React.Dispatch<React.SetStateAction<Person[]>>;
+  loading: boolean;
+  error: string | null;
+};
+
+export const UsersContext = createContext<UsersContextType>({
+  people: [],
+  setPeople: () => {},
+  loading: false,
+  error: null,
+});
+
+type Props = {
+  children: React.ReactNode;
+};
+
+export const UsersProvider: React.FC<Props> = ({ children }) => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+
+  useEffect(() => {
+    const fetchPeople = async () => {
+      setLoading(true);
+      try {
+        const data = await getPeople();
+
+        setPeople(data);
+      } catch {
+        setError(ErrorMessage.peopleLoadingError);
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    fetchPeople();
+  }, []);
+
+  return (
+    <UsersContext.Provider value={{ people, setPeople, loading, error }}>
+      {children}
+    </UsersContext.Provider>
+  );
+};
+
+export const usePeople = () => useContext(UsersContext);
```

Update `usePeople` to throw when the context is not provided. This makes misuse obvious and avoids silent failures. Replace the current export with e.g.: `export const usePeople = () => { const ctx = useContext(UsersContext); if (!ctx) throw new Error('usePeople must be used inside UsersProvider'); return ctx; }`